File: 백준/Silver/2075. N번째 큰 수/N번째 큰 수.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        PriorityQueue<Integer> pq = new PriorityQueue<>();

        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());

            for (int j = 0; j < N; j++) {
                int num = Integer.parseInt(st.nextToken());

                if (pq.size() < N) {
                    pq.offer(num);
                } else {
                    if (pq.peek() < num) {
                        pq.poll();
                        pq.offer(num);
                    }
                }
            }
        }

        System.out.println(pq.peek());
    }
}

File: 백준/Silver/2075. N번째 큰 수/N번째 큰 수.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.PriorityQueue
import java.util.StringTokenizer


fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val N = br.readLine().toInt()

    val pq = PriorityQueue<Int>()
    
    repeat(N){
        val st = StringTokenizer(br.readLine())
        repeat(N){
            val num = st.nextToken().toInt()
            if(pq.size < N){
                pq.offer(num)
            }else{
                if(pq.peek() < num){
                    pq.poll()
                    pq.offer(num)
                }
            }
        }
    }
    
    println(pq.peek())
}